Objective-C/Internal/Replicator/CBLRemoteDatabase+Internal.h

```diff
@@ -64,11 +64,12 @@ NS_ASSUME_NONNULL_BEGIN
 
 @end
 
-@interface ConnectedClientUpdateDocumentContext : ConnectedClientContext
+@interface ConnectedClientPutDocumentContext : ConnectedClientContext
 @property (nonatomic) NSString* docID;
 @property (nonatomic) FLSliceResult docBody;
 @property (nonatomic) BOOL isDeleted;
 @property (nonatomic) void(^docUpdateCompletion)(CBLDocument*, NSError*);
+@property (nonatomic) void(^docDeleteCompletion)(NSError*);
```

Comment:
I'm not sure if it is required to have separated completion block for update and delete by marking the returned CBLDocument* as nullable. If needed, you can add isDeleted flag to the block as well (but don't do it if it's not required). 

Also if docID property is nullable, can you add nullable to the docID as well? To simplify it, the docID can be required even for the deletion. 

To simplify, you can have only one constructor and you can add isDeleted only if that is required such as  

```Objective-C
- (instancetype) initWithRemoteDB: (CBLRemoteDatabase*)db
                                                 docID: (nullable NSString*)docID
                                             docBody: (FLSliceResult)docBody
                                             isDeleted: (BOOL)isDeleted
                                         completion: (void(^)(nullable CBLDocument*, nullable NSError*))completion;
```